src/index.js

```diff
@@ -0,0 +1,111 @@
+import { registerPlugin } from '@wordpress/plugins';
+import { PluginSidebar, PluginSidebarMoreMenuItem } from '@wordpress/edit-site';
+import { blockDefault } from '@wordpress/icons';
+import { Fragment, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { Button, MenuGroup, MenuItem, PanelBody, TextControl } from '@wordpress/components';
+import { store as noticesStore } from '@wordpress/notices';
+import { useDispatch } from '@wordpress/data';
+
+const GlobalStylesProvider = wp.editSite.GlobalStylesProvider;
+
+const StyleVariations = () => {
+	const [ variationName, setVariationName ] = useState( '' );
+	const [ canReset, onReset ] = wp.editSite.useGlobalStylesReset();
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	async function createVariation() {
+		try {
+			const response = await apiFetch( {
+				path: '/create-block-theme/v1/variation/' + variationName,
+			} );
+
+			// Clear global styles customizations
+			onReset();
+
+			// Reload variations
+			// We need to invalidate and maybe refetch __experimentalGetCurrentThemeGlobalStylesVariations(),
+			if ( response.req.status === 200 ) {
+				createErrorNotice( __( 'Variation created successfully' ), { type: 'snackbar' } );
+			}
+
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while creating the site export.' );
+			createErrorNotice( errorMessage, { type: 'snackbar' } );
+		}
+	}
+
+	return (
+		<PanelBody>
+			<MenuGroup label={ __( 'Create style variation' ) }>
+				<TextControl onChange={ ( value ) => setVariationName( value ) } placeholder={ __( 'Variation name' ) }></TextControl>
```

Comment:
May be a short description can be added to give the context of the feature.

`Save user changes as a new style variation of the current theme`

Reply:
I wonder if there's a tooltip component or something like that in GB so we don't add a lot fo text while still conveying all the information for the action

Reply:
Crashing for me too.